server/routes/appointments/appointmentsController.ts

```diff
@@ -451,16 +463,56 @@ export default class AppointmentsController {
         formError = data.error
         userInputData = req.body
       } else {
-        const updatedAppointment = await this.interventionsService.recordSupplierAssessmentAppointmentAttendance(
-          accessToken,
-          referralId,
-          data.paramsForUpdate
-        )
-        const redirectPath =
-          updatedAppointment.sessionFeedback?.attendance?.attended === 'no' ? 'check-your-answers' : 'behaviour'
-        return res.redirect(
-          `/service-provider/referrals/${referralId}/supplier-assessment/post-assessment-feedback/${redirectPath}`
-        )
+        let basePath
+        let redirectPath
+        if (draftBookingId) {
+          const fetchResult = await this.fetchDraftAppointmentOrRenderMessage(req, res)
+          if (fetchResult.rendered) {
+            return
+          }
+          const { draft } = fetchResult
+          const draftAppointment = draft.data as DraftAppointment
+          if (draftAppointment) {
+            if (draftAppointment.sessionFeedback) {
+              draftAppointment.sessionFeedback.attendance.attended = data.paramsForUpdate.attended!
+              draftAppointment.sessionFeedback.attendance.additionalAttendanceInformation =
+                data.paramsForUpdate.additionalAttendanceInformation!
+            } else {
+              draftAppointment.sessionFeedback = {
+                attendance: {
+                  attended: data.paramsForUpdate.attended!,
+                  additionalAttendanceInformation: data.paramsForUpdate.additionalAttendanceInformation!,
+                },
+                behaviour: {
+                  behaviourDescription: null,
+                  notifyProbationPractitioner: null,
+                },
+                submitted: false,
+                submittedBy: null,
+              }
+            }
+          } else {
+            throw new Error('Draft appointment data is missing.')
```

Comment:
this is a situation we never expect to be in, if it happens, we would probably need a more detailed log message to debug why.